Tests/RoutingTests/SheetManageableTests.swift

```diff
@@ -32,11 +32,23 @@ final class SheetManageableTests: XCTestCase {
         router.dismissSheet()
         XCTAssertNil(router.sheet)
     }
+    
+    func testPresentOnDismissingAction() {
+        
+        router.presentSheet(.settings, onDismiss: { })
+                
+        router.dismissSheet()
+        
+        XCTAssertNotNil(router.onDismiss)
+        XCTAssertNil(router.sheet)
```

Comment:
What I meant was, have a duplicate test like this one, but after these asserts, add:
```swift
router.presentSheet(.settings)
XCTAssertNil(router.onDismiss)
```